hack/cluster/cluster.go

```diff
@@ -65,6 +67,9 @@ func main() {
 	rand.Seed(time.Now().UnixNano())
 
 	if err := run(context.Background(), log); err != nil {
+		if oDataError, ok := err.(msgraph_errors.ODataErrorable); ok {
+			spew.Dump(oDataError.GetError())
+		}
```

Comment:
we're going to `log.Fatal(err)` on this anyway.  Should we just reassign error to a new error instead of spewing it out? 

What does an `ODataErrorable` type look like?  Do we need to spew it out, or are other fields helpful within it.?

Reply:
See the [Error resource type](https://learn.microsoft.com/en-us/graph/errors#error-resource-type) docs.  You have to cast it to this type to get any useful information out: error codes, meaningful descriptions, etc.

The standard `Error` type you get back from MS Graph function calls always just unhelpfully says "[error status code received from the API](https://github.com/microsoft/kiota-abstractions-go/blob/84ff2533063d4f3d1fff75ce4b56ec60be21584a/api_error.go#L12-L18)".  So I spew the OData details into the log to help with debugging.